### PR TITLE
restore: fix missing colum infos when restore from checkpoint

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Fix missing column infos when restore from checkpoint

### What is changed and how it works?
- Write Chunk ColumnPermutation to checkpoint db (file/mysql)  when insert or update chunk checkpoints.
- When restore from checkpoint, also restore column infos for parser from stored  `ColumnPermutation`

NOTE: this fix can't handle following case:

table schema:
```
create table t(a varchar(32), b int);
```

data.sql
```
insert into t (a, b) values 
("test1", 1),
("test2", 2);

insert into t (a) values 
("test3"),
("test4");
```

If multi insert statements contains different columns number for their order are different, Mable recover from checkpoint will still fail because we don't update columns infos on the fly. 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

Related changes

### CODE REVIEW

**Comment by glorv** [comment]
@kennytm PTAL

File: lightning/checkpoints/checkpoints.go [hunk 1]
```diff
@@ -1293,3 +1313,19 @@ func (cpdb *FileCheckpointsDB) DumpEngines(context.Context, io.Writer) error {
...
+
+func intSlice2Int32Slice(s []int) []int32 {
+	res := make([]int32, len(s))
+	for _, i := range s {
+		res = append(res, int32(i))
+	}
+	return res
+}
+
+func int32Slice2IntSlice(s []int32) []int {
```
Comment (kennytm) [inline lightning/checkpoints/checkpoints.go:1325]: this function is used nowhere.

File: tests/checkpoint_columns/run.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+set -euE
+
+# Populate the mydumper source
+DBPATH="$TEST_DIR/cppk.mydump"
+TABLE_COUNT=1
+CHUNK_COUNT=50
+
+mkdir -p $DBPATH
+echo 'CREATE DATABASE cp_tsr;' > "$DBPATH/cp_tsr-schema-create.sql"
+PARTIAL_IMPORT_QUERY='SELECT 0'
```
Comment (kennytm) [inline tests/checkpoint_columns/run.sh:25]: what is `$PARTIAL_IMPORT_QUERY`? used nowhere.

File: tests/checkpoint_columns/run.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+set -euE
+
+# Populate the mydumper source
+DBPATH="$TEST_DIR/cp.mydump"
+
+mkdir -p $DBPATH
+echo 'CREATE DATABASE cp_tsr;' > "$DBPATH/cp_tsr-schema-create.sql"
+echo "CREATE TABLE tbl(i TINYINT PRIMARY KEY, j INT);" > "$DBPATH/cp_tsr.tbl-schema.sql"
+echo "INSERT INTO tbl (i) VALUES (1),(2);" > "$DBPATH/cp_tsr.tbl.sql"
```
Comment (kennytm) [inline tests/checkpoint_columns/run.sh:24]:
```suggestion
echo "INSERT INTO tbl (j,i) VALUES (3,1),(4,2);" > "$DBPATH/cp_tsr.tbl.sql"
```

File: tests/checkpoint_columns/run.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+check_contains "count(*): 1"
+
+# restart lightning from checkpoint, the second line should be written successfully
+export GO_FAILPOINTS=
+set +e
+run_lightning -d "$DBPATH" --backend tidb --enable-checkpoint=1 2> /dev/null
+set -e
+
+run_sql 'SELECT count(*) FROM `cp_tsr`.tbl'
+check_contains "count(*): 2"
```
Comment (kennytm) [inline tests/checkpoint_columns/run.sh:47]:
```sh
run_sql 'SELECT j FROM `cp_tsr`.tbl WHERE i = 2'
check_contains 'j: 4'
```

**Comment by kennytm** [comment]
PTAL @3pointer

**Pull request merged**